### PR TITLE
Learnable inducing points and seed vectors

### PR DESCRIPTION
**Pull request by rvinas**

### CODE REVIEW

**Comment by arrigonialberto86** [comment]
I see that in the previous implementation `random.normal` returned a `Tensor`, while changing this to `Variable` effectively makes this a trainable parameter

**Comment by rvinas** [comment]
Exactly. I believe both the seed vectors and inducing points are trainable in the paper, right?

Many thanks for the implementation!

**Comment by michaelpoluektov** [comment]
From the [official PyTorch implementation](https://github.com/juho-lee/set_transformer/blob/master/modules.py):

```
class PMA(nn.Module):
    def __init__(self, dim, num_heads, num_seeds, ln=False):
        super(PMA, self).__init__()
        self.S = nn.Parameter(torch.Tensor(1, num_seeds, dim))
        nn.init.xavier_uniform_(self.S)
        self.mab = MAB(dim, dim, dim, num_heads, ln=ln)

    def forward(self, X):
        return self.mab(self.S.repeat(X.size(0), 1, 1), X)
```
       
They're using a trainable `nn.Parameter`.
